core/finality-grandpa/src/environment.rs

```diff
@@ -650,12 +667,15 @@ where
 		self.update_voter_set_state(|voter_set_state| {
 			let mut completed_rounds = voter_set_state.completed_rounds();
 
+			// NOTE: future integration will store the prevote and precommit index
```

Comment:
This reads like a TODO : )

Reply:
Yes, although it's already being worked on in an open PR (https://github.com/paritytech/substrate/pull/2611). Should I link to it here?